### PR TITLE
Changed to deoplete for vim8

### PR DESCRIPTION
**Pull request by robglarsen**

### CODE REVIEW

**Comment by lfilho** [comment]
Hi, thanks for your PR.

Please elaborate more why this change must happen. Any trade-offs? Any current limitations of deoplete (last I heard it was still under development). Anything missing from it? Why should we do it? Thanks!

**Comment by nandalopes** [comment]
**From Shougu/neocomplete:**

_Note: This plugin is not compatible with above Vim 8.2.1066! https://github.com/vim/vim/commit/bd84617d1a6766efd59c94aabebb044bef805b99_

_Note: Active development on neocomplete.vim has stopped. The only future changes will be bug fixes._

_Please see Deoplete.nvim._

**Comment by robglarsen** [comment]
Sorry I don't really know these were the changes that I did seems to work
now - it didn't before

File: vim/vundles/vim-improvements.vundle [hunk 1]
```diff
@@ -1,6 +1,8 @@
 Bundle "AndrewRadev/splitjoin.vim"
 Bundle "Raimondi/delimitMate"
-Bundle "Shougo/neocomplete.git"
+Bundle 'Shougo/deoplete.nvim'
+Bundle 'roxma/nvim-yarp'
```
Comment (lfilho) [inline vim/vundles/vim-improvements.vundle:4]: Why are these 2 extra dependencies needed?

File: vim/settings/neocomplete.vim [hunk 2]
```diff
@@ -1,34 +0,0 @@
-" neocomplete
-" Next generation completion framework.
-
-let g:acp_enableAtStartup = 0
-let g:neocomplete#enable_at_startup = 1
-let g:neocomplete#enable_camel_case = 1
-let g:neocomplete#enable_smart_case = 1
```
Comment (lfilho) [inline vim/settings/neocomplete.vim:7]: Are all configurations in this file present in deoplete? I don't see them explicit set in your PR and we don't want to introduce regressions in behavior for the users of this repo

**Review by lfilho — COMMENTED** [review]
Added a couple of comments more. It doesn't seem it's ready to merge yet. Thanks

File: vim/settings/neocomplete.vim (hunk 2 above)
Comment (robglarsen) [reply vim/settings/neocomplete.vim:7]: Again the details are here 
https://github.com/Shougo/deoplete.nvim

Comment (lfilho) [reply vim/settings/neocomplete.vim:7]: Sorry, that doesn't answer my question. You're removing configurations that might effect all users of this dotfiles repo. Pointing to deoplete's repo doesn't answer my question and the inherent risk of removing those configs. There's also another question above pending an answer :)

Thank you

**Comment by fishc44** [comment]
I think she was going to be there at the first time but we had her go for the kids my mom was going out for dinner with me.

**Comment by fishc44** [comment]
Apple apple pie. Apple apple peeler apple apple juice apple apple cider apple apple peeler apple apple pie.

File: vim/settings/neocomplete.vim (hunk 2 above)
Comment (daxgames) [reply vim/settings/neocomplete.vim:7]: Looks like a more complete `deoplete.vim` would look like:

```
let g:deoplete#enable_at_startup = 1

call deoplete#custom#option({
\ 'camel_case': v:true,
\ 'smart_case': v:true,
\ })
```

There does not appear to be a `let g:acp_enableAtStartup = 0` setting for `deoplete`.

Not sure if this is correct but it is what I am using with this PR.

**Comment by lfilho** [comment]
> Not sure if this is correct but it is what I am using with this PR.

So we need to make sure it is.

In short, this repo/dotfiles is used by thousands of people, for more than a decade, so we don't want to introduce breaking changes like how people do completion, or run the risk of results coming in a different order, etc. So please elaborate how you are making sure this won't happen. Until then we can't merge such risky PR

**Comment by daxgames** [comment]
Not arguing but I switched to `deoplete` because I would rather have something that works with newer vim releases than sticking with something that only works with an outdated vim release that does not work with other plugins that have left it behind.

I wonder how many of the thousands n of users you mentioned have found another way like I did.

Heck even `deoplete` went to bugfix releases only a week and a half ago.

**Comment by lfilho** [comment]
@daxgames regardless what to migrate to, which replacement is better, etc (I personally switched to NeoVim + coc on my own fork) that's beyond the point. When creating a PR we should first do our dilligence and ensure our users will have the same experience as before while calling their auto complete unless impossible.

There's a bunch of configurations / customizations being deleted in this PR without any explanation or assurance how they would work with the new one. I can't, as a maintainer of such large repo, accept a PR in this state 😇. We'd just be replacing one half-baked state with another.

Obs: wow, didn't know deoplete got deprecated already. Seems like Shougo is never satisfied with his plugins and is always recreating them haha. From neocomplcache to neocomplete to deoplete and now ddc. Yikes.

## So, next steps to this PR (basically what should be there since the beggining):

- [ ] Ensure all the existing neocomplete key mappings, configurations and customizations are mapped/transferred to deoplete accordingly. 
- [ ] Explanations on how you tested the above

@daxgames @robglarsen are you able to help out?

**Comment by daxgames** [comment]
Like I said I was not arguing and my additions were my limited attempt at help.  Honestly I never gave much thought to neocomplete config in `yadr` until I noticed it was not there because I upgraded to a vim version that broke it.

**Comment by nandalopes** [comment]
There's another problem: _active development on deoplete.nvim has stopped. The only future changes will be bug fixes._

**Comment by robglarsen** [comment]
This method is now outdated

**Pull request closed without merging**